app/code/Magento/Captcha/Plugin/CheckCaptchaOnStorefront.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Copyright 2024 Adobe
+ * All Rights Reserved.
+ */
+declare(strict_types=1);
+
+namespace Magento\Captcha\Plugin;
+
+use Magento\Captcha\Helper\Data as HelperCaptcha;
+use Magento\Customer\Block\Account\AuthenticationPopup;
+use Magento\Customer\Model\Context;
+use Magento\Framework\App\Http\Context as HttpContext;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Store\Model\ScopeInterface;
+
+/**
+ * Check need captcha for authentication popup
+ */
+class CheckCaptchaOnStorefront
+{
+    /**
+     * CheckCaptchaOnStorefront constructor
+     *
+     * @param HelperCaptcha $helper
+     * @param HttpContext $httpContext
+     * @param ScopeConfigInterface $scopeConfig
+     */
+    public function __construct(
+        private readonly HelperCaptcha $helper,
+        private readonly HttpContext   $httpContext,
+        private readonly ScopeConfigInterface $scopeConfig
+    ) {
+    }
+
+    /**
+     * Remove template when login or disable captcha storefront
+     *
+     * @param AuthenticationPopup $subject
+     * @param string $result
+     * @return string
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function afterGetTemplate(
+        AuthenticationPopup $subject,
+        ?string $result
+    ): string {
+        // Check the value for recaptcha_frontend/type_for/customer_login
+        $recaptchaType = $this->scopeConfig->getValue(
+            'recaptcha_frontend/type_for/customer_login',
+            ScopeInterface::SCOPE_WEBSITE
+        );
+
+        if ($this->isLoggedIn() || (!$this->helper->getConfig('enable') && $recaptchaType === null)) {
```

Comment:
I guess the logic about enabled form for custom login should be added at place, where we configure captcha fields in the customer login form.

I checked - currently it's configured using layout XML, so I guess it has to be moved from XML to some PHP code that will add needed parameters.

I guess it should be something like this:

https://github.com/magento/magento2/blob/c5614048eaaff80c1d4786b391e0c3cc55c7417c/app/code/Magento/Checkout/Block/Onepage.php#L78-L85
https://github.com/magento/magento2/blob/c5614048eaaff80c1d4786b391e0c3cc55c7417c/app/code/Magento/Paypal/etc/frontend/di.xml#L197-L203
https://github.com/magento/magento2/blob/c5614048eaaff80c1d4786b391e0c3cc55c7417c/app/code/Magento/Paypal/Block/PayLater/LayoutProcessor.php#L20-L68